File: c/fibonacci.c
unsigned int fibonacci(unsigned int n) {
    unsigned int a = 0;
    unsigned int b = 2;  // FIXME! correct value is 1

    switch (n) {
        case 0:
        case 1:
            return a;
        case 2:
            return b;
        default:
            break;
    }

    for (int i=0; i<n-2; i++) {
        unsigned int next = a + b;
        a = b;
        b = next;
    }
    return b;
}

File: python/fibonacci.py
#!/usr/bin/env python3
import sys

def fibonacci(n):
    a = 0
    b = 2  # FIXME! correct value is 1

    if n == 0 or n == 1:
        return a
    elif n == 2:
        return b

    for i in range(n-2):
        a, b = b, a + b
    return b


def main():
    args = sys.argv
    n = 5
    if 1 < len(args):
        s = args[1]
        if s.isdigit():
            n = int(s)

    for i in range(1, n+1):
        print(fibonacci(i))


if __name__ == "__main__":
    main()
